clang/test/dpct/memory_management_restricted.cu

```diff
@@ -418,3 +418,13 @@ void foo2() {
   cudaFree(a1);
   cudaFree(a2);
 }
+
+void foo3() {
+  float *a, *b;
+  // CHECK: auto flag = 0;
```

Comment:
For the current test case,  auto "flag=0" can be removed , please consider further refine in seperate PR.